middlewares/authenticator.go

```diff
@@ -76,9 +88,20 @@ func parserBasicUsers(users types.Users) (map[string]string, error) {
 	return userMap, nil
 }
 
-func parserDigestUsers(users types.Users) (map[string]string, error) {
+func parserDigestUsers(digest *types.Digest) (map[string]string, error) {
+	var userStrs []string
+	if digest.UsersFile != "" {
+		var err error
+		if userStrs, err = getUserStringsFromFile(digest.UsersFile); err != nil {
+			return nil, err
+		}
+	}
+	userStrs = append(digest.Users, userStrs...)
 	userMap := make(map[string]string)
-	for _, user := range users {
+	for _, user := range userStrs {
+		if user == "" {
+			continue
+		}
```

Comment:
The reading logic is identical to the part above, right? Could we move it straight into `getUserStringsFromFile` (maybe with adjusted function signature)?